### PR TITLE
New Quest API and kingdom fix

### PR DESCRIPTION
**Pull request by RyadaProductions**
+ added a new quest api
* fixed kingdomofmiscellania plugin

The quest API is for a plugin i am working on.
At the same time i took the liberty to fix the Throne of Miscellania plugin as well.

### CODE REVIEW

**Comment by Infinitay** [comment]
Here's the thing, they're not all `VarPlayer`, some are Varbits so this API addition would be incorrect

**Comment by RyadaProductions** [comment]
I have split the quests between `VarPlayer`  and `VarBits`, and improved the quest completion logic.
All people have to do to use the API is `Quests.ELEMENTAL_WORKSHOP_I.isCompleted(client)`  as an example.

File: runelite-api/src/main/java/net/runelite/api/Quests.java [hunk 1]
```diff
@@ -0,0 +1,411 @@
+package net.runelite.api;
```
Comment (MESLewis) [inline runelite-api/src/main/java/net/runelite/api/Quests.java]: Copyright

**Comment by Adam-** [comment]
I don't know of a good way to independently verify any of these are correct.

**Comment by Juzzed** [comment]
Tested with max QP. Works.

**Comment by RyadaProductions** [comment]
Please don't merge yet. Working on adding miniquests as well currently.

**Comment by RyadaProductions** [comment]
All API changes i wanted to implement should be in there now.
I added all the mini-quests as well and the option to see if a quest has been started as well.
This should provide more options to use this API.

**Comment by MagicfTail** [comment]
This is working on an account with all quests and miniquests completed

**Comment by Asymons** [comment]
Can this be looked at again? May need quest api for achievement diaries.

**Comment by trimbe** [comment]
@Adam- These are pulled directly from scripts 1357 (f2p), 1358 (p2p), and 1359 (mini), but I'm not sure how you'd go about automating the verification.

We could also just call those scripts from Java for the API but I don't think we gain a lot from doing that considering how rarely we need quest completion states in the client.

**Comment by Adam-** [comment]
Quest api was added in #7219

**Pull request closed without merging**